src/WebJobs.Extensions.DurableTask/Listener/TaskNonexistentActivityShim.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using DurableTask.Core;
+using DurableTask.Core.Common;
+using DurableTask.Core.Exceptions;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Listener
+{
+    internal class TaskNonexistentActivityShim : TaskActivity
+    {
+        private readonly DurableTaskExtension config;
+        private readonly string activityName;
+        private readonly string message;
+
+        public TaskNonexistentActivityShim(
+            DurableTaskExtension config,
+            string activityName,
+            string message)
+        {
+            this.config = config;
+            this.activityName = activityName;
+            this.message = message;
+        }
+
+        public override string Run(TaskContext context, string input)
+        {
+            Exception exceptionToReport = new FunctionFailedException(this.message);
+
+            throw new TaskFailureException(
```

Comment:
We may just be able to throw a different exception since I think the DTFx layer will wrap the exception for us (but I could be wrong).

Reply:
Ok, so looking at the normal `TaskActivityShim`, I see that we throw `TaskFailureException()` there as well, so I understand that. I guess my only remaining question is if there is a reason we use `FunctionFailedException` as the inner exception, instead of InvalidOperationException or similar?

Reply:
There's no specific reason. Would it be better to use `InvalidOperationException`, `FunctionInvocationException`, etc.?

Reply:
I think this is probably fine honestly.